### ISSUE TITLE
MP3 decoder add padding

### ISSUE THREAD

**Issue by adefossez** [pytorch/audio#1500]
## 🐛 Bug

Many MP3 encoders add padding at the beginning / end of audio files. See for instance
https://lame.sourceforge.io/tech-FAQ.txt

That means if you take a wavefile, encode it to mp3, and load it with torchaudio, it will actually be a bit longer than the original wav. I'm not sure if there is a perfect way to handle that, and if the amount can depend on the encoder. `ffmpeg` typically removes that padding automatically.

## To Reproduce

```python
import subprocess as sp
import torchaudio as ta
import torch
from demucs.audio import AudioFile

sr = 44100
t = torch.arange(sr * 4).float() / sr
f = torch.cos(2 * 3.14 * 440 * t)[None]
ta.save('cos.wav', f, sr)
sp.run(["lame", "-b320", "cos.wav"])
o, _ = ta.load("cos.mp3")
print(o.shape, f.shape, o2.shape)
# If you have demucs installed you can run the following, which is based on FFmpeg
# o2 = AudioFile("cos.mp3", channels=1).read()
# print(o2.shape)
```

## Expected behavior

I would expect `f` and `o` to have the same size, but in fact, `o` is 2160 samples longer than `f`. When using a ffmpeg based reader, I do get the same shape.

Now I don't know how ffmpeg handles that, or if there might be corner cases. It does seem a bit weird that going from wav to mp3 and back to wav would change the size with torchaudio, and I think the majority of audio software do remove this padding.

I'm also not entirely clear on why 2160 samples. From the lame FAQ it is not that clear.

## Environment

 - PyTorch Versi on (e.g., 1.0): 1.8.1
 - torchaudio is 0.8.1
 - OS (e.g., Linux): Mac
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version: NA
 - GPU models and configuration: NA
 - Any other relevant information:

**Comment by mthrok** [comment]
Hi @adefossez 

Thanks for the suggestion. I was aware of this but somehow thought this is a part of the specification. I will take a look into this.

**Comment by adefossez** [comment]
Thanks :)
It is surprising that this is not an internal details of the encoder/decoder, and it doesn't seem that well documented :(

**Comment by adefossez** [comment]
Thanks for the update @mthrok. I am a bit sad to see the support for mp3 go away without ffmpeg installed but its good that is this solved now !
